File: Exercism.io/python/sum-of-multiples/sum_of_multiples.py

class SumOfMultiples(object):
    def __init__(self, *nn):
        self.factors = nn or (3,5)
    
    def to(self,n):
        return sum(x for x in range(1,n) if self._is_multiple(x))
    
    def _is_multiple(self, n):
        return any((n % x == 0) for x in self.factors)

File: Exercism.io/elixir/sum-of-multiples/sum_of_multiples.exs
defmodule SumOfMultiples do

  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  
  The default factors are 3 and 5.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors \\ [3, 5]) do
    1..limit-1
      |> Enum.filter(&(has_factor?(&1, factors)))
      |> Enum.sum
  end

  defp has_factor?(n, factors) do
    factors
      |> Enum.map(&(rem(n, &1) == 0))
      |> Enum.any?
  end

end
